2025 11 updates

lucor opened the pull request:
This PR updates:
- Go to v1.25.3
- Zig to v0.15.2

lucor committed:
feat(base): image update

This commit updates Go to v1.25.3, Zig to v0.15.2. Additionally move to
fyne tools v1.7.0

lucor committed:
feat(darwin-sdk-extractor): update to be compatible with zig 0.15.2

lucor committed:
fix(base): rollback to fyne 2.5.5

To use fyne tools an updated version of fyne-cross is needed.
Rollback to fyne CLI.

ThePhilderbeast commented on base/Dockerfile:
```diff
@@ -1,9 +1,9 @@
-ENV GO_AMD64_CHECKSUM="bbca37cc395c974ffa4893ee35819ad23ebb27426df87af92e93a9ec66ef8712"
-ENV GO_ARM64_CHECKSUM="124ea6033a8bf98aa9fbab53e58d134905262d45a022af3a90b73320f3c3afd5"
-ENV ZIG_VERSION=0.14.1
+ENV GO_VERSION=1.25.3
```
Go has now moved to 1.25.6 it would be worth updating this before this is merged.

lucor committed:
chore(base): update Go to v1.25.10

lucor committed:
ci: bump GitHub Actions to latest major versions

Update actions/setup-go from v3 to v6 to fix Go 1.19.x download
failures caused by deprecated fallback URLs and outdated manifests.
Update actions/checkout to v6 across all workflows for consistency.
